kv/raftstore/peer_storage.go

```diff
@@ -261,7 +261,7 @@ func (ps *PeerStorage) clearExtraData(newRegion *metapb.Region) {
 	if bytes.Compare(oldStartKey, newStartKey) < 0 {
 		ps.clearRange(newRegion.Id, oldStartKey, newStartKey)
 	}
-	if bytes.Compare(newEndKey, oldEndKey) < 0 {
+	if bytes.Compare(newEndKey, oldEndKey) < 0 || len(oldEndKey) == 0 {
```

Comment:
what if both `newEndKey` and `oldEndKey` are empty?

Reply:
If two both are empty, it shouldn't call `clearRange()`.  
Maybe change like this?
```suggestion
	if bytes.Compare(newEndKey, oldEndKey) < 0 || (len(oldEndKey) == 0 && len(newEndKey) != 0) {
```

Reply:
looks good